### ISSUE TITLE
In the integration tests, SpotBugs does not recognize the FindBugs NonNull annotation

### ISSUE THREAD

**Issue by mewore** [spotbugs/spotbugs#1656]
For this code, I would expect SpotBugs to warn about `NP_NONNULL_RETURN_VIOLATION` while it actually doesn't produce any warning:
```java
public class Issue1656 {
    @edu.umd.cs.findbugs.annotations.NonNull
    public String testReturnNullInNonNullMethod() {
        return null; // expected NP_NONNULL_RETURN_VIOLATION
    }
}
```

When I replaced the annotation with the JSR305 one `javax.annotation.Nonnull` or with the Checker Framework one `org.checkerframework.checker.nullness.qual.NonNull`, it worked.

If this is intended, I think it should be mentioned as a big warning in the [contributing guidelines](https://github.com/spotbugs/spotbugs/blob/5564ed751f4b203d873244febcc2f28ba59adfde/.github/CONTRIBUTING.md).

Here is the test ([f5a9bb5ff6fbf98fa676e0ef5d1627113796e4ab](https://github.com/mewore/spotbugs/commit/f5a9bb5ff6fbf98fa676e0ef5d1627113796e4ab)):
* https://github.com/mewore/spotbugs/blob/f5a9bb5ff6fbf98fa676e0ef5d1627113796e4ab/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/nullness/Issue1656Test.java
* https://github.com/mewore/spotbugs/blob/master/spotbugs-tests/src/test/java/edu/umd/cs/findbugs/nullness/Issue1656Test.java
* https://github.com/mewore/spotbugs/blob/master/spotbugsTestCases/src/java/ghIssues/Issue1656.java

And a part of its output:
```
Method invocation:    1: invokespecial[183](3) 1
	Invoking: java.lang.Object.<init>()
	Prototype method: class=java.lang.Object, method=public void <init>() [RuntimeVisibleAnnotations]
	Target method set: [java.lang.Object.<init>()]
Analyzing ghIssues.Issue1656.testReturnNullOnNonNullMethod()


Expected: Iterable containing exactly <1> BugInstance with:
bugType="NP_NONNULL_RETURN_VIOLATION",
     but: was <[]>
```

**Comment by KengoTODA** [comment]
Annotations provided by FindBugs/SpotBugs aren't well-defined so nobody can explain its expected behavior. For instance, [`@Nullable` exists but does nothing](https://spotbugs.readthedocs.io/en/stable/annotations.html#edu-umd-cs-findbugs-annotations-nullable). JSR305 has similar problem too.


If you expect bytecode analysis tool detects potential NPEs, I personally suggest you to use other tools such as [NullAway](https://github.com/uber/NullAway) or [Checkerframework](https://checkerframework.org/manual/#nullness-checker), they use well-documented nullness annotation and provide rules based on it. For me it's too difficult to fix SpotBugs' implementation with keeping backward compatibilities... because we have no document that explains the backward compatibility (spec) we need to keep.
